rules/enforce-eslint-mandatory-prop.js

```diff
@@ -0,0 +1,46 @@
+const hasProp = require("jsx-ast-utils/hasProp");
+
+module.exports = {
+  /** @type {import('eslint').Rule.RuleMetaData} */
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Enforce passing a mandatory prop to a component",
+    },
+    schema: [
+      {
+        type: "object",
+        additionalProperties: {
+          type: "object",
+          additionalProperties: {
+            type: "object",
+            additionalProperties: {
+              type: "boolean",
+            },
+          },
+        },
+      },
+    ],
+  },
+  /** @type {import('eslint').Rule.RuleModule['create']} */
```

Comment:
Should we type the whole object instead?

Reply:
Done